sharktuner/boo_tuner/boo_tuner.py

```diff
@@ -0,0 +1,410 @@
+        ]
+    )
+
+    return compile_args
+
+
+def tune_boo_benchmark(
+    benchmark_path: Path,
+    args: argparse.Namespace,
+    path_config: libtuner.PathConfig,
+    root_logger: logging.Logger,
+    summary_handler: logging.Handler,
+    starter_td_spec: Path | None,
+) -> Path | None:
+    """Tune a single BOO benchmark dispatch."""
+    args.input_file = benchmark_path
+    # Only use starter spec if it exists and the file is present.
+    if starter_td_spec and starter_td_spec.exists():
+        args.starter_td_spec = starter_td_spec
+    else:
+        args.starter_td_spec = None
+
+    logging.info("Generating candidate tuning specs...")
+    with common.TunerContext(logger=root_logger) as tuner_context:
+        tuner_context.logger.addHandler(summary_handler)
+        boo_tuner = BooTuner(tuner_context)
+        candidates = libtuner.generate_candidate_specs(args, path_config, boo_tuner)
+        logging.info(f"Stored candidate tuning specs in {path_config.specs_dir}")
+
+        logging.info("Compiling dispatch candidates...")
+        boo_tuner.compile_flags = ["--compile-from=executable-sources"]
+        compiled_candidates = libtuner.compile(args, path_config, candidates, boo_tuner)
+
+        logging.info("Benchmarking compiled dispatch candidates...")
+        boo_tuner.benchmark_flags = [
+            "--input=1",
+            "--benchmark_repetitions=3",
+        ]
+        top_candidates = libtuner.benchmark(
+            args,
+            compiled_candidates,
+            boo_tuner,
+            args.boo_tuner_num_dispatch_candidates,
+            args.boo_dispatch_benchmark_timeout_mins,
+        )
+
+        if not top_candidates:
+            logging.critical("No tuning candidates performed better than the baseline.")
+            return None
+
+        logging.info(f"Top dispatch candidates: {top_candidates}")
+        for id in top_candidates:
+            logging.info(f"{boo_tuner.candidate_trackers[id].spec_path.resolve()}")
+
+        # Save the best (first) tuning spec to output file.
+        best_candidate_id = top_candidates[0]
+        best_spec_path = boo_tuner.candidate_trackers[best_candidate_id].spec_path
+        shutil.copy(best_spec_path, args.output_td_spec)
+        logging.info(f"Saved best tuning spec to: {args.output_td_spec}")
+        print(f"Saved best tuning spec to: {args.output_td_spec}")
+
+    return args.output_td_spec
+
+
+def process_boo_command(
+    cli_args: list[str],
+    args: argparse.Namespace,
+    boo_path_config: BooPathConfig,
+    root_logger: logging.Logger,
+    starter_td_spec: Path | None,
+    boo_runtime,
+    get_launchable,
+    BooOpRegistry,
+) -> Path | None:
+    """Process a single BOO command through compilation and tuning."""
+    sig = BooOpRegistry.parse_command(cli_args, ignore_unhandled_args=True)
+    if sig is None:
+        raise ValueError(f"Boo op registry failed to parse '{shlex.join(cli_args)}'.")
+
+    # Setup temporary directory.
```

Comment:
```suggestion
    # Set up temporary directory.
```

setup is a noun, set up is a phrasal verb